EFCore.SqlServer.HierarchyId.Test/Migrations/Design/CSharpMigrationOperationGeneratorTest.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Adapted From: https://github.com/dotnet/efcore/blob/v3.1.0/test/EFCore.Design.Tests/Migrations/Design/CSharpMigrationOperationGeneratorTest.cs
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
```

Comment:
Hmm, not sure we're allowed to relicense that under the MIT...

Reply:
Yeah, that is a big question.  Everything in the TestUtilities folder was copied from dotnet/efcore (in an effort to keep it fairly easy to merge to dotnet/efcore).  Should these migration tests be removed?

Reply:
Merging this back into EF Core seems like it'll be an even bigger nightmare. 😅 Let's cross that bridge when we get there.

Personally, I think most of this testing infrastructure is probably unnecessary for this repo. Maybe just simplify it to what's absolutely needed and call it an original work. 🤷‍♂️

Reply:
Migration tests have now been simplified